ERCS/erc-7815.md

```diff
@@ -0,0 +1,263 @@
+    /// @return tokens Array of token contract addresses
+    function getTokens(
+        bytes32 poolId
+    ) external returns (address[] memory tokens);
+}
+```
+
+Retrieves tokens for a given pool. This function is OPTIONAL.
+
+- MUST return all tokens that are tradeable by the pool.
+
+## Rationale
+
+### A short overview of functions needed in order routing
+
+This ERC aims to make it easy for anyone to discover and use defi liquidity. Especially solvers, searchers, and aggregators (here collectively termed “routers”).
+
+Here are common and important pieces of different routing algorithms, that inspired the design of this ERC:
+
+**Marginal Price**: The marginal price is the first derivative of the swap (amount out) function at a particular trade amount. The spot price, i.e. the current price, is the marginal price at trade amount 0. Having easy access to the marginal price at different trade amounts is essential to:
+
+- **Filter pools**: Pools with the best spot price have the best "instantaneous swap rate" for the token pair – making them a good candidate to include in a swap route. Second derivatives help judge even better which pool will provide the highest amount out for the swap (area under the swap curve).
+- **Determine swap amounts**: Marginal prices are also very good for sizing steps in iterative algorithms: Swap on the pool with the best marginal price until it reaches the marginal price of the second best pool, then split between both until you reach the marginal price of the third best pool etc.
+- **Curve interpolation**: Solvers can also use marginal prices to construct analytical interpolations of price functions that might not have simple analytical representations. Sample at multiple points and then build an interpolation. Analytical interpolation are useful to find analytical solutions to trade and split amounts – before using iterative approaches for fine-tuning the splits.
+- **Price after swap equivalence**: Besides discontinuous price functions and discrete gas costs, optimal solutions to route splitting must equalise the marginal prices of all pools included in the split.
+- **Convex optimisation**: Framing the optimal routing problem as a convex optimisation problem, which has gained more popularity recently, also relies on fast and accurate access to marginal prices.
+From these uses in different routing approaches we derived the need for:
+- **price**: A price function that can provide the spot price (marginal price at 0 amount in) and the marginal price *after* any trade amount.
+- **priceDerivative**: Derivatives of marginal prices, i.e. second derivatives of the swap function, are very useful for choosing pools, and for analytically interpolating the swap function.
+- **swap to price**: A direct analytical derivation of the trade amount needed to move the liquidity pool to a specific marginal price speeds up routers, by avoiding iterative queries to the price function.
```

Comment:
This seems like general background information, or at least I'm having a hard time linking it to decisions made in your proposal. The rationale section should be used to explain why you made a particular technical choice within the specification. For example, you could explain why you chose an enum over a string for capabilities.